utils/lit/lit/formats/taef.py

```diff
@@ -141,30 +147,49 @@ def execute(self, test, litConfig):
         if litConfig.noExecute:
             return lit.Test.PASS, ''
 
-        out, err, exitCode = executeCommandForTaef(
-            cmd, env = test.config.environment)
+        out, exitCode = executeCommandForTaef(
+            cmd, env=test.config.environment)
 
-        if exitCode:
-            skipped = 'Failed=0, Blocked=0, Not Run=0, Skipped=1'
-            if skipped in out:
-                return lit.Test.UNSUPPORTED, ''
+        return getTestResult(out, exitCode), out
+
+
+def getTestResult(out, exitCode):
+    unselected = 'The selection criteria did not match any tests.'
+    if unselected in out:
+        return lit.Test.UNSUPPORTED
+
+    # TAEF's exit code cannot reliably indicate if a test was skipped, failed or
+    # passed. The summary string instead is used to determine this.
+
+    # Example summary string:
+    # Summary: Total=1, Passed=1, Failed=0, Blocked=0, Not Run=0, Skipped=0
+    regex = r"Summary: Total=\d+, Passed=(\d+), Failed=(\d+), Blocked=(\d+), Not Run=(\d+), Skipped=(\d+)"
+
+    # Get the last occurance of the summary in the output
+    match = None
+    for m in re.finditer(regex, out):
+        match = m
+
+    if not match:
```

Comment:
Do you mean to indent this into the for loop?

Reply:
finditer only yields matches, so the only way for match to end up as None would be if finditer didn't yield anything.